File: Homework_2/app/src/main/java/com/example/homework_2/MainActivity.java
package com.example.homework_2;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

   private TextView SumU1;
   private TextView SumU2;
   private final Simulation simulation = new Simulation(this);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SumU1 = findViewById(R.id.sum_U1);
        SumU2 = findViewById(R.id.sum_U2);
    }

    @SuppressLint("SetTextI18n")
    public void runTheSimulation(View view) throws Exception {
        int total_Budget = simulation.runSimulation(simulation.loadU1(simulation.loadItems("Phase-1.txt")));
        total_Budget += simulation.runSimulation(simulation.loadU1(simulation.loadItems("Phase-2.txt")));
        SumU1.setText(total_Budget + " million $");

        total_Budget = simulation.runSimulation(simulation.loadU2(simulation.loadItems("Phase-1.txt")));
        total_Budget += simulation.runSimulation(simulation.loadU2(simulation.loadItems("Phase-2.txt")));
        SumU2.setText(total_Budget + " million $");

    }
}

File: SpaceShip_Kotlin/app/src/main/java/com/example/spaceship_kotlin/MainActivity.kt
package com.example.spaceship_kotlin

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private var SumU1: TextView? = null
    private var SumU2: TextView? = null
    private var simulation = Simulation(this)



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        SumU1 = findViewById(R.id.sum_U1)
        SumU2 = findViewById(R.id.sum_U2)
    }

    @SuppressLint("SetTextI18n")
    fun runTheSimulation(view: View){
       var totalBudget = simulation.runSimulation(simulation.loadU1(simulation.loadItems("Phase-1.txt")))
        totalBudget += simulation.runSimulation(simulation.loadU1(simulation.loadItems("Phase-2.txt")))
        SumU1!!.text = "$totalBudget million $"

       totalBudget = simulation.runSimulation(simulation.loadU2(simulation.loadItems("Phase-1.txt")))
        totalBudget += simulation.runSimulation(simulation.loadU2(simulation.loadItems("Phase-2.txt")))
        SumU2!!.text = "$totalBudget million $"
    }
}